### PR TITLE
feat(Zigbee): Support for sleepy device + Power Source and battery level + Humidity sensor cluster

### LINKED ISSUE #10405: Zigbee Implementation: Deep Sleep example
### Related area

Zigbee

### Hardware specification

ESP32-C6

### Is your feature request related to a problem?

Hi,
thanks for the tremendous and fast implementation and merge of the Zigbee library!

I am wondering if there is already an easy way at this point to set up a battery powered ESP32(-C6) endpoint utilising deep sleep?

Or are there higher obstacles to pass to realize deep sleep/short wake up times for e.g. sending sensor readings?

Thanks again very much, best regards,
Joost

### Describe the solution you'd like

Perhaps one of the examples could be enhanced slightly to include deep sleep with periodic wake up.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

### LINKED ISSUE #10430: Zigbee Implementation: Battery Management, Charger Management, and Power Management
### Related area

Zigbee

### Hardware specification

ESP32-H2

### Is your feature request related to a problem?

Related to https://github.com/espressif/arduino-esp32/issues/10135 (Implement Zigbee in Arduino) and the sleep modes feature request for sleepy Zigbee End Devices ->https://github.com/espressif/arduino-esp32/issues/10405

At the bare minimum Zigbee need to include "Power Source" (i.e. mains power or battery powered) and battery level information.

But also wondering if can include advanced features for Battery Management, Charger Management, and Power Management?

If not as default then as one or more examples?

FYI, related to this I recently also read that Arduino has released an open-source Arduino Power Management library for Arduino Pro that could maybe be ported to arduino-esp32 for ESP32? It includes advanced rea--time battery management and battery monetoring + battery charger manager, as well as "sleep mode" (deep sleep) and "standby mode" (light sleep), with profiles for specific development boards:

* https://github.com/arduino-libraries/Arduino_PowerManagement/
  * https://github.com/arduino-libraries/Arduino_PowerManagement/tree/main/examples
  
For more information on how to use that library see their documentation:

* https://github.com/arduino-libraries/Arduino_PowerManagement/blob/main/docs

More on that new Arduino PowerManagement library here:

* https://www.cnx-software.com/2024/09/26/arduino-power-management-library-arduino-pro-modules/

### Features
* Monitor current and average battery metrics (voltage, current, percentage)
* Monitor battery health metrics (temperature, reported capacity)
* Monitor and control charging
* Save significant amounts of power by sending the boards into Sleep (only available for Portenta C33 for now) and Standby modes
* Toggle and set voltages on different power rails of the board
  * On Nicla Vision
    * Enable and disable the external power rail
    * Enable and disable the camera power rail
    * Change voltage on external power rail
  *   On Portenta C33
    * Enable and disable the external power rail
    * Enable and disable the communication power rail (ESP32, secure element)
    * Change voltage on the external power rail
  * On Portenta H7
    * Enable and disable the external power rail
    * Change voltage on external power rail

### Describe the solution you'd like

Maybe a Power Management library library for ESP32 and Zigbee examples for ESP32-H2, ESP32-H4, ESP32-C5, and ESP32-C6?

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
This PR includes:

- Support for sleepy device (only deep-sleep is currently possible).
- Support for selecting power source for the device. If battery is selected, the remaining percentage can be updated.
- Support for Humidity cluster, which was added as optional for the Temperature sensor as there is no HA Device type for humidity sensor.
- New example `Zigbee_Temp_Hum_Sensor_Sleepy`, which demonstrates all the new functionalities
- Mirror fixes and improvements

## Tests scenarios
Tested with the example `Zigbee_Temp_Hum_Sensor_Sleepy` on both C6/H2 which connects to the Zigbee network.
Coordinator used - SONOFF ZBDongle-E which is configured in the Home Assistant which runs on VM.

## Related links
Closes #10405
Closes #10430

### CODE REVIEW

File: libraries/Zigbee/src/ZigbeeEP.cpp [hunk 1]
```diff
@@ -5,16 +5,32 @@
 #if SOC_IEEE802154_SUPPORTED
 
 #include "esp_zigbee_cluster.h"
+#include "zcl/esp_zigbee_zcl_power_config.h"
 
-uint8_t ZigbeeEP::_endpoint = 0;
+#ifdef __cplusplus
```
Comment (me-no-dev) [inline libraries/Zigbee/src/ZigbeeEP.cpp:10]: this is always defined. You are in CPP file

File: libraries/Zigbee/src/ZigbeeEP.cpp [hunk 2]
```diff
@@ -5,16 +5,32 @@
...
+extern "C"
+{
+#endif
+
+extern void zb_set_ed_node_descriptor(bool power_src, bool rx_on_when_idle, bool alloc_addr);
```
Comment (me-no-dev) [inline libraries/Zigbee/src/ZigbeeEP.cpp:15]:
```cpp
extern "C" void zb_set_ed_node_descriptor(bool power_src, bool rx_on_when_idle, bool alloc_addr);
```

**Comment by me-no-dev** [comment]
Please create a `Zigbee.h` header that includes all other headers and would be the only required header to include

File: libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  // When all EPs are registered, start Zigbee in End Device mode
+  Zigbee.begin(&zigbeeConfig, false);
+
+  // Wait for Zigbee to start
+  while(!Zigbee.isStarted()){
+    delay(100);
+  }
+
+  // Delay 5s to allow establishing connection with coordinator, needed for sleepy devices
+  delay(5000);
```
Comment (SuGlider) [inline libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino:105]: this time could be a #define ?
It make it easier for the user to finf and change it.

File: libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+void loop() {
+  // Checking button for factory reset
+  if (digitalRead(BUTTON_PIN) == LOW) {  // Push button pressed
+    // Key debounce handling
+    delay(100);
+    int startTime = millis();
+    while (digitalRead(BUTTON_PIN) == LOW) {
+      delay(50);
+      if ((millis() - startTime) > 3000) {
```
Comment (SuGlider) [inline libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino:116]: factoryReset time coud be a #define or const uint32_t xxx
That would make it configurable.

**Review by SuGlider — COMMENTED** [review]
LGTM. Just a few suggestion.

File: libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino (hunk 3 above)
Comment (P-R-O-C-H-Y) [reply libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino:105]: I don't think it's good to change it in general. In many tests. this delay act as perfect, so the data are sent 100% times.
With smaller delay it was behaving weirdly.

**Review by lucasssvaz — APPROVED** [review]
LGTM

**Review by SuGlider — APPROVED** [review]

**Comment by ggtimtom** [comment]
Hello P-R-O-C-H-Y,

thank you for this great work!
I want to connect the Zigbee Sleepy End Device Example to Zigbee2MQTT. Is this possible?

regards, tt

**Pull request merged**